### ISSUE TITLE
Only "target" local posts

### ISSUE THREAD

**Issue by timgrossmann** [InstaPy/InstaPy#3]
#### Would be nice if you could e.g. only like and comment and follow, that where posted near you
It could search through the tags and look at the posted location (if given) to determine if it's near your location

Could look somewhat like this:
```python
.set_only_locals(True) #default False
.set_locals(['Stuttgart', 'Ludwigsburg', 'Bad Canstatt']) 
.like_by_location([...])
```

- [ ] implement location lookup (tags and, if give, location)
- [ ] implement 'near cities' which gets the adjacent and near cities of the given location
- [ ] check the posts if they where taken near user
- [x] like posts by a location

**Comment by littleswany** [comment]
@timgrossmann I like the idea but if you are using a widely used hashtag would the percentage go photos from the given locations not be very low so therefore have a large number of skips?

Is there a way within IG to search by location and use that as an alternative to hashtags. This would work well for small businesses trying to interact with local IG users.

**Comment by timgrossmann** [comment]
@littleswany Yes exactly this is the idea.

Well, you could simply use the hashtags of that area e.g. Stuttgart to start with.
And the set_only_locals should run as long as there are n posts found with the locations given.

Still have to look into IG if a search by location is possible. Thank you for the update on this!

**Comment by littleswany** [comment]
@timgrossmann I can work on something as you described using the hashtag of a location. I will start once the clean up has been merged to the develop branch.

From a quick google search it docent look much like you can search by location, I will look into it in a bit more depth and report my findings.

**Comment by timgrossmann** [comment]
@littleswany Perfect, thank you!
I think it would be best for that feature if there was like a check if they only used the location hashtag or if they actually tagged their location with the location feature... As long as you can see it on the Web interface...

**Comment by littleswany** [comment]
@timgrossmann Yeah I'm with you, I am pretty sure you can retrieve it using JS just as you do with the username. I will post any update here.

**Comment by littleswany** [comment]
@timgrossmann You can browse posts by location on the browser but it would require the user to input the location code (https://www.instagram.com/explore/locations/), and then IstaPy can function as if it was a hashtag page.

Using this method would replace the use of a hashtag rather than supplement it as the other suggests.

**Comment by timgrossmann** [comment]
@littleswany Holy s***, this is perfect! We could even add a "topic" parameter which could be something like restaurant or clothing and only pick out the ones that are of that kind 👍 

@littleswany Thank you so much

**Comment by ghost** [comment]
@timgrossmann @littleswany Hey guys what is the status of this? I would like to contribute to this project if possible.

**Comment by timgrossmann** [comment]
@imakethefish Still open... 
Feel free to contribute!

**Comment by timgrossmann** [comment]
Partly implemented at #276
